miio/cooker.py

```diff
@@ -0,0 +1,706 @@
+        idle:                              ['waiting',      '0001', 'null',             '29',     '60',      '-1',   '60',  '0607', '05040f', '00030017',   '0100', 'ffffffffffff011effff010000001d1f']
+        quickly preheat:                   ['running',      '0001', '00000000ff', '031e0b23',     '60',      '-1',   '60',  '0607', '05040f', '00030017',   '0100', 'ffffffffffff011effff010000001d1f']
+        absorb water at moderate temp:     ['running',      '0001', '02000000ff', '031e0b23',     '54',      '-1',   '60',  '0607', '05040f', '00030017',   '0100', 'ffffffffffff011effff010002013e23']
+        absorb water at moderate temp:     ['running',      '0001', '02000000ff', '031e0b23',     '48',      '-1',   '60',  '0607', '05040f', '00030017',   '0100', 'ffffffffffff011effff010002013f29']
+        operate at full load to boil rice: ['running',      '0001', '03000000ff', '031e0b23',     '39',      '-1',   '60',  '0607', '05040f', '00030017',   '0100', 'ffffffffffff011effff010003055332']
+        operate at full load to boil rice: ['running',      '0001', '04000000ff', '031e0b23',     '35',      '-1',   '60',  '0607', '05040f', '00030017',   '0100', 'ffffffffffff011effff010004026460']
+        operate at full load to boil rice: ['running',      '0001', '06000000ff', '031e0b23',     '29',      '-1',   '60',  '0607', '05040f', '00030017',   '0100', 'ffffffffffff011effff010006015c64']
+        high temperature gelatinization:   ['running',      '0001', '07000000ff', '031e0b23',     '22',      '-1',   '60',  '0607', '05040f', '00030017',   '0100', 'ffffffffffff011effff010007015d64']
+        temperature gelatinization:        ['running',      '0001', '0a000000ff', '031e0b23',     '2',       '-1',   '60',  '0607', '05040f', '00030017',   '0100', 'ffffffffffff011effff01000a015559']
+        meal is ready:                     ['autokeepwarm', '0001', '1000000000', '031e0b23031e', '1',      '750',   '60',  '0207', '05040f', '00030017',   '0100', 'ffffffffffff011effff01000000535d']
+        """
+        self.data = data
+
+    @property
+    def mode(self) -> OperationMode:
+        """Current operation mode."""
+        return OperationMode(self.data['func'])
+
+    @property
+    def menu(self) -> int:
+        """Selected recipe id."""
+        return int(self.data['menu'], 16)
+
+    @property
+    def stage(self) -> Optional[CookingStage]:
+        """Current stage if cooking."""
+        stage = self.data['stage']
+        if len(stage) == 10 and stage.hexdigits:
+            return CookingStage(stage)
+
+        return None
+
+    @property
+    def temperature(self) -> Optional[int]:
+        """
+        Current temperature, if idle.
+
+        Example values: *29*, 031e0b23, 031e0b23031e
+        """
+        value = self.data['temp']
+        if len(value) == 2 and value.isdigit():
+            return int(value)
+
+        return None
```

Comment:
Exception instead of returning None here too? (and below also, for error cases)

Reply:
If the cooker isn't cooking it returns the room temperature here (property: "temp").

Reply:
Ok, what are the values otherwise? Those hex-encoded ones? Should this be printing a warning when used and the temperature is not available, and point out what to do instead of it?

Reply:
Note: Discussed at discord.